nodeApps/createNpmRelease.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# Create a bundle for uploading to a github release and npm
+# Requires the following env vars to be set: GITHUB_USER, GITHUB_API_TOKEN
+
+set -e
+
+if [ "${GITHUB_USER}" == "" ] || [ "${GITHUB_API_TOKEN}" == "" ]; then
+  echo "ERROR: GitHub credentials not set."
+  exit 1
+fi
+
+if [ "${NPM_AUTH}" == "" ] || [ "${NPM_NAME}" == "" ] || [ "${NPM_EMAIL}" == "" ]; then
+  echo "ERROR: NPM credentials not set."
+  exit 1
+fi
+
+# Check if we are on the correct branch to release from
+RELEASE_BRANCH="master"
+if [ "${TRAVIS_BRANCH}" != "" ]; then
+  CURRENT_BRANCH=${TRAVIS_BRANCH}
+else
+  CURRENT_BRANCH=`git rev-parse --abbrev-ref HEAD`
+fi
+echo "Current branch: ${CURRENT_BRANCH}"
+if [ "${CURRENT_BRANCH}" != "${RELEASE_BRANCH}" ]; then
+  echo "ERROR: Not on ${RELEASE_BRANCH} branch"
+  exit 2
+fi
+
+# Check if we have not release this version already
+git pull
+LAST_TAG=`git tag --list | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort | tail -n 1`
+echo "Last tag: ${LAST_TAG}"
+THIS_VERSION=`cat package.json | grep version | head -n 1 | cut -d '"' -f 4`
+echo "This version: ${THIS_VERSION}"
+if [ "v${THIS_VERSION}" == "${LAST_TAG}" ]; then
+  echo "WARNING: Version not updated"
+  exit 0
+fi
+
+git checkout ${RELEASE_BRANCH}
```

Comment:
we probably do not need to do this any more as we are not commiting to the branch in this script, i can remove it.